internal/pkg/cli/local_run.go

```diff
@@ -0,0 +1,336 @@
+	ecsLocalClient     ecsLocalClient
+	deployStore        deployedEnvironmentLister
+	envVersionGetter   func(string) (versionGetter, error)
+}
+
+func newLocalRunOpts(vars localRunVars) (*localRunOpts, error) {
+	sessProvider := sessions.ImmutableProvider(sessions.UserAgentExtras("local run"))
+	defaultSess, err := sessProvider.Default()
+	if err != nil {
+		return nil, err
+	}
+
+	store := config.NewSSMStore(identity.New(defaultSess), ssm.New(defaultSess), aws.StringValue(defaultSess.Config.Region))
+	deployStore, err := deploy.NewStore(sessProvider, store)
+	if err != nil {
+		return nil, err
+	}
+
+	ws, err := workspace.Use(afero.NewOsFs())
+	if err != nil {
+		return nil, err
+	}
+	ecsLocalClient := ecs.New(defaultSess)
+	opts := &localRunOpts{
+		localRunVars: vars,
+
+		deployedWkld:       []string{},
+		wkldDeployedToEnvs: make(map[string][]string),
+		prompt:             prompt.New(),
+		store:              store,
+		ws:                 ws,
+		ecsLocalClient:     ecsLocalClient,
+		deployStore:        deployStore,
+	}
+	opts.envVersionGetter = func(envName string) (versionGetter, error) {
+		envDescriber, err := describe.NewEnvDescriber(describe.NewEnvDescriberConfig{
+			App:         opts.appName,
+			Env:         envName,
+			ConfigStore: opts.store,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("new environment compatibility checker: %v", err)
+		}
+		return envDescriber, nil
+	}
+	return opts, nil
+}
+
+func (o *localRunOpts) Validate() error {
+	if o.appName == "" {
+		return errNoAppInWorkspace
+	}
+	if _, err := o.store.GetApplication(o.appName); err != nil {
+		return fmt.Errorf("get application %s: %w", o.appName, err)
+	}
+	return nil
+}
+
+func (o *localRunOpts) Ask() error {
+	if err := o.validateOrAskWorkloadName(); err != nil {
+		return err
+	}
+	if err := o.validateOrAskEnvName(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (o *localRunOpts) Execute() error {
+	taskDef, err := o.ecsLocalClient.TaskDefinition(o.appName, o.envName, o.wkldName)
+	if err != nil {
+		return fmt.Errorf("get task definition: %w", err)
+	}
+
+	awsSession := session.Must(session.NewSessionWithOptions(
+		session.Options{
+			SharedConfigState: session.SharedConfigEnable,
+		}))
+
+	_, err = o.getDecryptedSSMSecrets(taskDef, awsSession)
+	if err != nil {
+		fmt.Println("unable to get secrets from SSM store")
+	}
+
+	_, err = o.getDecryptedSecretManagerSecrets(awsSession)
+	if err != nil {
+		fmt.Println("unable to get secrets from Secret Manager")
+	}
+
+	return nil
+}
+
+func (o *localRunOpts) getDecryptedSSMSecrets(taskDef *awsecs.TaskDefinition, awsSession *session.Session) ([]ssmSecret, error) {
+	var SSMSecretsList []ssmSecret
+	secrets := taskDef.Secrets()
+	ssmClient := ssm.New(awsSession)
+	for _, secret := range secrets {
+		resp, err :=
+			ssmClient.GetParameter(&ssm.GetParameterInput{
+				Name:           aws.String(secret.ValueFrom),
+				WithDecryption: aws.Bool(true),
+			})
+		if err != nil {
+			return nil, err
```

Comment:
This is an example where we _should_ wrap the error.
```suggestion
			return nil, fmt.Errorf("get SSM parameter %q: %w", aws.String(secret.ValueFrom), err)
```